physics/euler_solver_body.hpp

```diff
@@ -151,6 +202,85 @@ EulerSolver<PrincipalAxesFrame>::AngularMomentumAt(
   };
 }
 
+template<typename InertialFrame, typename PrincipalAxesFrame>
+typename EulerSolver<InertialFrame, PrincipalAxesFrame>::AttitudeRotation
+EulerSolver<InertialFrame, PrincipalAxesFrame>::AttitudeAt(
+    AngularMomentumBivector const& angular_momentum,
+    Instant const& time) const {
+  Rotation<PrincipalAxesFrame, ℬₜ> const 𝒫ₜ =
+      Compute𝒫ₜ(moments_of_inertia_, angular_momentum);
+
+  Time const Δt = time - initial_time_;
+  Angle ψ;
+  switch (formula_) {
+    case Formula::i: {
+      Angle const φ = JacobiAmplitude(λ₃_ * Δt - ν_, mc_);
+      ψ = ψ_t_multiplier_ * Δt +
+          ψ_Π_multiplier_ * (EllipticΠ(φ, n_, mc_) - ψ_Π_offset_);
+      break;
+    }
+    case Formula::ii: {
+      Angle const φ = JacobiAmplitude(λ₁_ * Δt - ν_, mc_);
+      ψ = ψ_t_multiplier_ * Δt +
+          ψ_Π_multiplier_ * (EllipticΠ(φ, n_, mc_) - ψ_Π_offset_);
+      break;
+    }
+    case Formula::iii: {
+      Angle const angle = λ₂_ * Δt - ν_;
+      ψ = ψ_t_multiplier_ * Δt +
+          ψ_Π_multiplier_ *
+              (angle + n_ * std::log(n_ * Sinh(angle) - Cosh(angle)) * Radian -
+               ψ_Π_offset_);
+      break;
+    }
+    case Formula::Sphere: {
+      ψ = ψ_t_multiplier_ * Δt;
+      break;
+    }
+    default:
+      LOG(FATAL) << "Unexpected formula " << static_cast<int>(formula_);
+  };
+  Bivector<double, ℬₜ> const e₃({0, 0, 1});
+  Rotation<ℬₜ, ℬʹ> const 𝒴ₜ(ψ, e₃, DefinesFrame<ℬʹ>{});
+
+  return ℛ_ * 𝒴ₜ * 𝒫ₜ;
+}
+
+template<typename InertialFrame, typename PrincipalAxesFrame>
+Rotation<PrincipalAxesFrame,
+         typename EulerSolver<InertialFrame, PrincipalAxesFrame>::ℬₜ>
+EulerSolver<InertialFrame, PrincipalAxesFrame>::Compute𝒫ₜ(
+    R3Element<MomentOfInertia> const& moments_of_inertia,
+    Bivector<AngularMomentum, PrincipalAxesFrame> const& angular_momentum) {
+  auto const& m = angular_momentum;
+  auto const& m_coordinates = m.coordinates();
+
+  // Compute ṁ using the Euler equation.
+  auto const& I₁ = moments_of_inertia.x;
+  auto const& I₂ = moments_of_inertia.y;
+  auto const& I₃ = moments_of_inertia.z;
+  Bivector<Quotient<AngularMomentum, MomentOfInertia>, PrincipalAxesFrame> const
+      ω({m_coordinates.x / I₁, m_coordinates.y / I₂, m_coordinates.z / I₃});
+  Bivector<Variation<AngularMomentum>, PrincipalAxesFrame> const ṁ =
+      Commutator(m, ω) / Radian;
+
+  // If ṁ is constant in the principal axes frame, pick 𝒫ₜ = identity.
```

Comment:
`If m is constant`, or `If ṁ is 0`.
But how is this correct? If `𝒫ₜ` is the identity in coordinates, `𝒫ₜ(m_normalized)` is `e₃` (in `ℬₜ`) only if `m_normalized` is `e₃` (in `PrincipalAxesFrame`); but `m` can be constant for at least 6 different values of `m_normalized` (`±eᵢ` in `PrincipalAxesFrame`), and for many values if some moments of inertia are equal.

Reply:
Done, it gets a bit ugly.